From ce7b6aee657d8d7fd9740cd3ea9b695ea3e75d3e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Thu, 2 May 2024 11:33:44 +0200
Subject: [PATCH 591/841] feat(stm32mp21-fdts): add support for watchdog
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Support iwdg1 in stm32mp215f-dk board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 065194b1bdf60d837c856755ea662ad58118e02e feat(stm32mp21-fdts): add support for watchdog]
Change-Id: Icdb8d4ee63fad6eec1564c4e58bf3786e1e22996
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/380600
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 fdts/stm32mp211.dtsi    | 8 ++++++++
 fdts/stm32mp215f-dk.dts | 5 +++++
 2 files changed, 13 insertions(+)

diff --git a/fdts/stm32mp211.dtsi b/fdts/stm32mp211.dtsi
index 556fc7a09..4eccef39c 100644
--- a/fdts/stm32mp211.dtsi
+++ b/fdts/stm32mp211.dtsi
@@ -197,6 +197,14 @@
 				status = "disabled";
 			};
 
+			iwdg1: watchdog@44010000 {
+				compatible = "st,stm32mp1-iwdg";
+				reg = <0x44010000 0x400>;
+				clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
+				clock-names = "pclk", "lsi";
+				status = "disabled";
+			};
+
 			sdmmc1: mmc@48220000 {
 				compatible = "st,stm32mp25-sdmmc2", "arm,pl18x", "arm,primecell";
 				arm,primecell-periphid = <0x00353180>;
diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index efd6f33fa..85e0e8cf7 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -30,6 +30,11 @@
 	};
 };
 
+&iwdg1 {
+	timeout-sec = <32>;
+	status = "okay";
+};
+
 &usart2 {
 	pinctrl-names = "default", "idle", "sleep";
 	pinctrl-0 = <&usart2_pins_a>;
-- 
2.47.2

